### PR TITLE
Add Installation Guide  for PGMQ

### LINKED ISSUE #238: Installation guide for multiple platforms
```[tasklist]
### Tasks
- [ ] Guide for PGXN
- [ ] Guide for Windows
- [ ] Guide for Mac
```

### PR DESCRIPTION
**Pull request by Saturn225**
Closes #238
/claim #238

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -8,6 +8,26 @@ The fastest way to get started is by running the Tembo docker image, where PGMQ
...
+## PGXN Installation
+
+PGXN is a centralized network to make it easier to install and manage PostgreSQL extensions.
+
+You can install PGMQ directly using PGXN, the PostgreSQL Extension Network, which is a distribution system for open-source PostgreSQL extension libraries.
+
+Install PGMQ using PGXN
+
+```bash
+pgxn install pgmq
```
Comment (ChuckHend) [inline CONTRIBUTING.md:20]: Please add hyperlinks to pgxn, either link directly to the pgxn install instructions or put the instructions here.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -17,6 +37,57 @@ version 14 on ubuntu:
 sudo apt-get install postgres-server-dev-14
 ```
 
+## Platform-Specific Installation Instructions
+
+### Windows Installation (Using WSL)
```
Comment (ChuckHend) [inline CONTRIBUTING.md:42]: Can we provide instructions for Native windows instead? Instructions to install pgmq on WSL are the same as linux install instructions.

**Comment by Saturn225** [comment]
@ChuckHend Done!

**Comment by ChuckHend** [comment]
@Saturn225 thank you, looks good! Should be able to merge once we get a chance to review the windows example.

**Comment by ChuckHend** [comment]
closes https://github.com/tembo-io/pgmq/issues/316

**Comment by Saturn225** [comment]
> @Saturn225 thank you, looks good! Should be able to merge once we get a chance to review the windows example.

review windows example? @ChuckHend

**Comment by ChuckHend** [comment]
Yes I'd like one of us to walk through the windows example and see how it goes before accepting. I should be able to do this in the next couple days.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -17,13 +41,82 @@ version 14 on ubuntu:
...
+
+- Download the official PostgreSQL installer from [PostgreSQL Windows Downloads](https://www.postgresql.org/download/windows/).
+
+- Follow the installation process, ensuring you include pgAdmin and Command Line Tools
+
+- After installation, add the PostgreSQL binary directory (typically `C:\Program Files\PostgreSQL\<version>\bin`) to your system's `PATH` environment variable. This will allow access to PostgreSQL tools like `pg_config`.
+
+2. Install Build Tools (MinGW)
+
+- Download and install [MinGW]() from MinGW. Select the GCC compiler for C/C++
```
Comment (ChuckHend) [inline CONTRIBUTING.md:60]: @Saturn225 it looks like MinGW is missing the url.

**Review by ChuckHend — CHANGES_REQUESTED** [review]
Please add missing link in the MinGW install instructions.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -8,6 +8,30 @@ The fastest way to get started is by running the Tembo docker image, where PGMQ
...
+```
+
+Alternatively, you can manually install PGMQ by downloading and building from the source using the following commands:
+
+```
+curl -LO https://api.pgxn.org/dist/pgmq/1.4.2/pgmq-1.4.2.zip
+unzip pgmq-1.4.2.zip
+cd pgmq-1.4.2
+make
+sudo make install
```
Comment (theory) [inline CONTRIBUTING.md:32]: I would omit `sudo` here, and suggest instead that you add a comment up top that these commands should be run as the user who owns the Postgres install (e.g., often `root` on *nix systems), or use `sudo`.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -17,13 +41,82 @@ version 14 on ubuntu:
...
+
+- After installation, connect to PostgreSQL and create the pgmq extension:
+
+```sql
+CREATE EXTENSION pgmq cascade;
+```
+
+### Mac Installation
+
+If you are using macOS, you can install PostgreSQL and build PGMQ using Homebrew.
```
Comment (theory) [inline CONTRIBUTING.md:90]: I don't think you need homebrew instructions. One can either use PGXN or download and build manually regardless of how Postgres was installed.

I think I'd suggest providing these categories for installation instruction, after saying the user must already have Postgres and build tools installed:

1. Use the `pgxn` CLI.
2. Download tarball from PGXN and build manually
3. Clone the Git repo and build manually
4. Windows

**Review by theory — CHANGES_REQUESTED** [review]

**Comment by Saturn225** [comment]
@theory Thanks for review! Will push the  review changes requested

File: CONTRIBUTING.md (hunk 5 above)
Comment (Saturn225) [reply CONTRIBUTING.md:90]: Yeah! It makes sense

File: CONTRIBUTING.md (hunk 3 above)
Comment (Saturn225) [inline CONTRIBUTING.md:60]:
```suggestion
- Download and install [MinGW](https://sourceforge.net/projects/mingw/) from MinGW. Select the GCC compiler for C/C++
```

File: CONTRIBUTING.md (hunk 4 above)
Comment (Saturn225) [inline CONTRIBUTING.md:32]:
```suggestion
make install
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -8,6 +8,30 @@ The fastest way to get started is by running the Tembo docker image, where PGMQ
...
+You can install the PGXN client by following the instructions provided in the [PGXN Client Installation Guide](https://pgxn.github.io/pgxnclient/install.html)
+
+Install PGMQ using PGXN
+
+```bash
+pgxn install pgmq
+```
+
+Alternatively, you can manually install PGMQ by downloading and building from the source using the following commands:
+
```
Comment (Saturn225) [inline CONTRIBUTING.md:26]:
```suggestion
> **Note:** Run these commands as the user who owns the PostgreSQL installation (often `root` on Unix-based systems) or use `sudo` if necessary.
```

File: CONTRIBUTING.md (hunk 4 above)
Comment (Saturn225) [reply CONTRIBUTING.md:32]: Done!

**Comment by Saturn225** [comment]
@ChuckHend @theory Have a another round of review on it. If any changes needed I'd like to incorporate them

File: CONTRIBUTING.md (hunk 3 above)
Comment (Saturn225) [reply CONTRIBUTING.md]: Added!

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -17,13 +42,82 @@ version 14 on ubuntu:
...
+- During installation, ensure the binaries are added to the `PATH`. MinGW is essential for compiling the `pgmq` extension on Windows.
+
+3.  Install `pgmq` from Source
+
+- Download the `pgmq` source:
+
+```bash
+curl -LO https://api.pgxn.org/dist/pgmq/1.4.2/pgmq-1.4.2.zip
+unzip pgmq-1.4.2.zip
+cd pgmq-1.4.2
```
Comment (ChuckHend) [inline CONTRIBUTING.md:71]: @Saturn225 , on windows, should this still be `bash`, and will the user have access to `curl` and `unzip`?

Comment (ChuckHend) [reply CONTRIBUTING.md:71]: If this is Powershell, lets make that clear.

Comment (ChuckHend) [reply CONTRIBUTING.md:71]: I'm a noob on Windows, and I'm getting stuck on this part 😹

Comment (Saturn225) [reply CONTRIBUTING.md:71]: @ChuckHend Yes that could make different if you go full with Windows Instructions

Comment (Saturn225) [reply CONTRIBUTING.md:71]:
```
   Invoke-WebRequest -Uri https://api.pgxn.org/dist/pgmq/1.4.2/pgmq-1.4.2.zip -OutFile pgmq-1.4.2.zip
   Expand-Archive -Path pgmq-1.4.2.zip -DestinationPath .\pgmq-1.4.2
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -17,13 +42,82 @@ version 14 on ubuntu:
...
+2. Install Build Tools (MinGW)
+
+- Download and install [MinGW](https://sourceforge.net/projects/mingw/) from MinGW. Select the GCC compiler for C/C++
+- During installation, ensure the binaries are added to the `PATH`. MinGW is essential for compiling the `pgmq` extension on Windows.
+
+3.  Install `pgmq` from Source
+
+- Download the `pgmq` source:
+
+```bash
```
Comment (Saturn225) [inline CONTRIBUTING.md:68]:
```suggestion
```powershell
   Invoke-WebRequest -Uri https://api.pgxn.org/dist/pgmq/1.4.2/pgmq-1.4.2.zip -OutFile pgmq-1.4.2.zip
   Expand-Archive -Path pgmq-1.4.2.zip -DestinationPath .\pgmq-1.4.2
   cd .\pgmq-1.4.2
```

File: CONTRIBUTING.md (hunk 7 above)
Comment (Saturn225) [inline CONTRIBUTING.md:71]:
```suggestion
```

**Comment by Saturn225** [comment]
@ChuckHend I have updated the installation steps, have a look once

**Review by ChuckHend — APPROVED** [review]

**Review by ChuckHend — APPROVED** [review]

**Comment by ChuckHend** [comment]
Waiting on @theory to respond they had requested changes.

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -8,6 +8,31 @@ The fastest way to get started is by running the Tembo docker image, where PGMQ
 docker run -d --name postgres -e POSTGRES_PASSWORD=postgres -p 5432:5432 quay.io/tembo/pgmq-pg:latest
 ```
 
+## PGXN Installation
+
+PGXN is a centralized network to make it easier to install and manage PostgreSQL extensions.
```
Comment (theory) [inline CONTRIBUTING.md:13]:
```suggestion
PGXN is a distributed network of extensions that make it easier to install and manage PostgreSQL extensions.
```

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -17,13 +42,82 @@ version 14 on ubuntu:
...
+
+3.  Install `pgmq` from Source
+
+- Download the `pgmq` source:
+
+```powershell
+   Invoke-WebRequest -Uri https://api.pgxn.org/dist/pgmq/1.4.2/pgmq-1.4.2.zip -OutFile pgmq-1.4.2.zip
+   Expand-Archive -Path pgmq-1.4.2.zip -DestinationPath .\pgmq-1.4.2
+   cd .\pgmq-1.4.2
+```
```
Comment (theory) [inline CONTRIBUTING.md:72]:
```suggestion
```powershell
Invoke-WebRequest -Uri https://api.pgxn.org/dist/pgmq/1.4.2/pgmq-1.4.2.zip -OutFile pgmq-1.4.2.zip
Expand-Archive -Path pgmq-1.4.2.zip -DestinationPath .\pgmq-1.4.2
cd .\pgmq-1.4.2
```
```

**Review by theory — APPROVED** [review]

**Pull request merged**